src/lib.rs

```diff
@@ -439,6 +579,45 @@ impl Navigator {
         }
     }
 
+    /// Reads all sensors and stores on a single structure.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use navigator_rs::{Navigator, SensorData};
+    /// use std::thread::sleep;
+    /// use std::time::Duration;
+    ///
+    /// let mut nav = Navigator::new();
+    /// nav.init();
+    ///
+    /// loop {
+    ///     let sensor_data: SensorData = nav.read_all();
+    ///     println!("---------------reading---------------");
+    ///     println!(
+    ///         "ADC values: {}, {}, {}, {}",
+    ///         sensor_data.adc.channel[0],
+    ///         sensor_data.adc.channel[1],
+    ///         sensor_data.adc.channel[2],
+    ///         sensor_data.adc.channel[3],
```

Comment:
Why not printing the structures directly ?

Reply:
done.

Reply:
That was not done, you could print:
`snsor_data.adc`, `nav.read_gyro` and etc.